.github/workflows/build-publish-mcr.yml

```diff
@@ -39,15 +44,20 @@ jobs:
     runs-on: ubuntu-latest
     needs: prepare-variables
     steps:
-    - uses: actions/checkout@v4
+    - name: Harden Runner
+      uses: step-security/harden-runner@eb238b55efaa70779f274895e782ed17c84f2895 # v2.6.1
+      with:
+        egress-policy: audit
+
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
       with:
         ref: ${{ needs.prepare-variables.outputs.release_tag }}
         
     - uses: ./.github/actions/install-deps
 
     # reference: https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-azure
     - name: Login to Azure
-      uses: azure/login@v1
+      uses: azure/login@cb79c773a3cfa27f31f25eb3f677781210c9ce3d # v1.6.1
```

Comment:
azure/login 1.6.0 added pre/post, which generates warnings in logs ([issue](https://github.com/Azure/login/issues/409)), could pin to 1.5.1 ...